### ISSUE TITLE
[BUG] Incorrect Pylint relative-beyond-top-level

### ISSUE THREAD

**Issue by 0az** [PyCQA/prospector#380]
**Describe the bug**
Prospector incorrectly reports `relative-beyond-top-level`.

**To Reproduce**
1. Create package `mcve` at `src/mcve`, including `src/mcve/__init__.py`
2. Create modules `foo` and `foobar` in the package as follows:
```
# foo.py
def print_foo():
	print('foo')
```
```
# foobar.py
from .foo import print_foo

def print_foobar():
	print_foo()
	print('bar')
```
2. Call prospector with `prospector src/mcve/foobar.py`
3. Prospector reports a false positive `relative-beyond-top-level`.

**Expected behavior**
Prospector mirrors Pylint behavior, and does not return any lint errors for `relative-beyond-top-level`.

**Screenshots**

```
Messages
========

src/mcve/foobar.py
  Line: 1
    pylint: relative-beyond-top-level / Attempted relative import beyond top-level package



Check Information
=================
         Started: 2020-05-28 13:44:24.195588
        Finished: 2020-05-28 13:44:24.347647
      Time Taken: 0.15 seconds
       Formatter: grouped
        Profiles: default, no_doc_warnings, no_test_warnings, strictness_medium, strictness_high, strictness_veryhigh, no_member_warnings
      Strictness: None
  Libraries Used: 
       Tools Run: dodgy, mccabe, pep8, profile-validator, pyflakes, pylint
  Messages Found: 1
```

Vs:

```
************* Module mcve.foobar
src/mcve/foobar.py:1:0: C0111: Missing module docstring (missing-docstring)
src/mcve/foobar.py:3:0: C0111: Missing function docstring (missing-docstring)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

**Environment (please complete the following information):**
 - OS: macOS
 - Tool: Pylint
 - Prospector version: 1.2.0
- Python version: 3.8.3

**Additional context**

**Comment by chocoelho** [comment]
@0az does it occur in 1.3.0.dev4? I updated pylint version in there. I'll add a test case for this situation nonetheless.

**Comment by 0az** [comment]
Verified fix with 1.3.0.dev4.
